### PR TITLE
SLCORE-897 Path should be converted to server path before submitting to the web API

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: medium-tests/src/test/java/mediumtest/tracking/IssueTrackingMediumTests.java [hunk 1]
```diff
@@ -713,6 +717,55 @@ void it_should_match_a_previously_tracked_issue_with_a_server_issue_when_binding
...
+    var connectionId = "connectionId";
+    var projectKey = "projectKey";
+    var server = newSonarCloudServer(orgKey)
+      .withProject(projectKey, project -> project
+        .withFile("server/path/prefix/pom.xml")
+          .withQualityProfile("qp")
+      )
+      .withQualityProfile("qp", qualityProfile -> qualityProfile.withLanguage("xml")
+        .withActiveRule("xml:S3421", rule -> rule.withSeverity(IssueSeverity.MAJOR)))
+      .withVersion("8.0.0.55884")
```
Comment (damien-urruty-sonarsource) [inline medium-tests/src/test/java/mediumtest/tracking/IssueTrackingMediumTests.java:742]: Why do you need a version for SonarCloud?

File: medium-tests/src/test/java/mediumtest/fixtures/ServerFixture.java [hunk 2]
```diff
@@ -158,6 +158,11 @@ public ServerBuilder withPlugin(String pluginKey, UnaryOperator<ServerPluginBuil
       return this;
     }
 
+    public ServerBuilder withVersion(String version) {
+      this.version = version;
+      return this;
+    }
+
```
Comment (damien-urruty-sonarsource) [inline medium-tests/src/test/java/mediumtest/fixtures/ServerFixture.java:165]: Normally this information is needed only in the SQ context, that's why it's only part of the `newSonarQubeServer` method and not a `with` as usual

**Review by damien-urruty-sonarsource — APPROVED** [review]
Some questions about the tests, else LGTM!

File: medium-tests/src/test/java/mediumtest/tracking/IssueTrackingMediumTests.java (hunk 1 above)
Comment (kirill-knize-sonarsource) [reply medium-tests/src/test/java/mediumtest/tracking/IssueTrackingMediumTests.java:742]: https://github.com/SonarSource/sonarlint-core/blob/55a175cddc96ad0cc015ec8ad13e7c102d7724b3/backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/LocalStorageSynchronizer.java#L54

This fails for the `null` version and blocks all further sync.

File: medium-tests/src/test/java/mediumtest/fixtures/ServerFixture.java (hunk 2 above)
Comment (kirill-knize-sonarsource) [reply medium-tests/src/test/java/mediumtest/fixtures/ServerFixture.java:165]: It's not the case with the current flow. We read the version from the storage during sync. See other reply.

**Pull request merged**